homeassistant/components/switch/tplink.py

```diff
@@ -60,3 +65,27 @@ def turn_on(self, **kwargs):
     def turn_off(self):
         """Turn the switch off."""
         self.smartplug.state = 'OFF'
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        _LOGGER.debug("Updating TP-Link energy meter data")
+
+        emeter_readings = self.smartplug.get_emeter_realtime()
```

Comment:
Indeed, `self.smartplug.state` is used to fetch data from the plug. I moved everything to `update` and the state and the meter's values are now stored as class variables.